diff --git a/sysdeps/unix/sysv/linux/loongarch/configure b/sysdeps/unix/sysv/linux/loongarch/configure
index 5636c344ba..2e86871a9f 100644
--- a/sysdeps/unix/sysv/linux/loongarch/configure
+++ b/sysdeps/unix/sysv/linux/loongarch/configure
@@ -184,10 +184,10 @@ lp64)
   test -n "$libc_cv_slibdir" ||
 case "$prefix" in
 /usr | /usr/)
-  libc_cv_slibdir='/liblp64'
-  libc_cv_rtlddir='/liblp64'
+  libc_cv_slibdir='/lib64'
+  libc_cv_rtlddir='/lib64'
   if test "$libdir" = '${exec_prefix}/lib'; then
-    libdir='${exec_prefix}/liblp64';
+    libdir='${exec_prefix}/lib64';
     # Locale data can be shared between 32-bit and 64-bit libraries.
     libc_cv_complocaledir='${exec_prefix}/lib/locale'
   fi
diff --git a/sysdeps/unix/sysv/linux/loongarch/configure.ac b/sysdeps/unix/sysv/linux/loongarch/configure.ac
index 900f5b6d04..aaf721e688 100644
--- a/sysdeps/unix/sysv/linux/loongarch/configure.ac
+++ b/sysdeps/unix/sysv/linux/loongarch/configure.ac
@@ -20,7 +20,7 @@ lp32)
   LIBC_SLIBDIR_RTLDDIR([liblp32], [liblp32])
   ;;
 lp64)
-  LIBC_SLIBDIR_RTLDDIR([liblp64], [liblp64])
+  LIBC_SLIBDIR_RTLDDIR([lib64], [lib])
   ;;
 esac
 
