diff --git a/sysdeps/unix/sysv/linux/loongarch/clone.S b/sysdeps/unix/sysv/linux/loongarch/clone.S
index 0b4eb5ea..4e897632 100644
--- a/sysdeps/unix/sysv/linux/loongarch/clone.S
+++ b/sysdeps/unix/sysv/linux/loongarch/clone.S
@@ -42,8 +42,8 @@ ENTRY (__clone)
 	/* The syscall expects the args to be in different slots.  */
 	or	a0, a2, zero
 	or	a2, a4, zero
-	or	a3, a5, zero
-	or	a4, a6, zero
+	or	a3, a6, zero
+	or	a4, a5, zero
 
 	/* Do the system call.  */
 	dli	a7,__NR_clone
diff --git a/sysdeps/unix/sysv/linux/loongarch/ipc_priv.h b/sysdeps/unix/sysv/linux/loongarch/ipc_priv.h
new file mode 100644
index 00000000..85646c44
--- /dev/null
+++ b/sysdeps/unix/sysv/linux/loongarch/ipc_priv.h
@@ -0,0 +1,26 @@
+/* Old SysV permission definition for Linux.  LoongArch version.
+   Copyright (C) 2020 Loongson Technology, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, see
+   <http://www.gnu.org/licenses/>.  */
+
+#include <sys/ipc.h>  /* For __key_t  */
+#include <kernel-features.h>
+
+#ifdef __ASSUME_SYSVIPC_DEFAULT_IPC_64
+# define __IPC_64      0x0
+#else
+# define __IPC_64      0x100
+#endif
diff --git a/sysdeps/unix/sysv/linux/loongarch/kernel-features.h b/sysdeps/unix/sysv/linux/loongarch/kernel-features.h
index d21c8246..d6c1c90a 100644
--- a/sysdeps/unix/sysv/linux/loongarch/kernel-features.h
+++ b/sysdeps/unix/sysv/linux/loongarch/kernel-features.h
@@ -19,9 +19,6 @@
 
 #include_next <kernel-features.h>
 
-#undef __ASSUME_CLONE_DEFAULT
-#define __ASSUME_CLONE_BACKWARDS 1
-
 /* No support for PI mutexes or robust futexes before 4.20.  */
 #if __LINUX_KERNEL_VERSION < 0x041400
 # undef __ASSUME_SET_ROBUST_LIST
-- 
2.27.0

